vapi/esx/settings/clusters/vms/transition.go

```diff
@@ -0,0 +1,226 @@
+type TransitionSpec struct {
+	// SourceCluster is the cluster to transition from.
+	SourceCluster types.ManagedObjectReference `json:"source_cluster"`
+
+	// Solution is the target desired solution specification in vLCM.
+	Solution *SolutionSpec `json:"solution"`
+}
+
+// EnableSpec contains fields that describe specification for enablement of EAM managed solution in vLCM.
+type EnableSpec struct {
+	// EamAgencyID is the identifier of the solution in EAM (EAM agency).
+	EamAgencyID string `json:"eam_agency_id"`
+
+	// Solution is the target desired solution specification in vLCM.
+	Solution *SolutionSpec `json:"solution"`
+}
+
+// MultiSourceEnableSpec contains fields that describe specification for enablement of multiple EAM managed solutions into single vLCM managed solution.
+// Supported only for solutions with deployment type DeploymentType#CLUSTER_VM_SET.
+type MultiSourceEnableSpec struct {
+	// EamAgencyIDs is the EAM Agency identifiers.
+	EamAgencyIDs []string `json:"eam_agency_ids"`
+
+	// Solution is the target desired solution specification in vLCM.
+	// The given SolutionSpec should not contain any AlternativeVmSpecs. See SourceAlternativeVmSpecs
+	// about how to configure AlternativeVmSpecs.
+	Solution *SolutionSpec `json:"solution"`
+
+	// SourceVmSelectionSpecs is the relation between System VMs and their respective VmSelectionSpecs.
+	// Provided VM IDs must be part of the solution being transitioned and
+	// must exist in the cluster where the solution is installed.
+	// Provided VmSelectionSpecs must be present in the applied desired state as part of the
+	// ClusterSolutionSpec#alternativeVmSpecs.
+	//
+	// If unset, no VmSelectionSpecs are applied on the source agencies' System VMs during the enablement.
+	SourceVmSelectionSpecs map[string]*VmSelectionSpec `json:"source_vm_selection_specs,omitempty"`
+
+	// ClusterModule to be reused for transitioned System VMs. Used to express
+	// VM-VM anti affinity relation between System VMs in the vSphere Cluster.
+	// The module must exist for the cluster where the solution is installed.
+	//
+	// If unset, no cluster module is reused. vLCM creates a new module if needed.
+	ClusterModule string `json:"cluster_module,omitempty"`
+}
+
+// Enable enables an EAM managed solution in vLCM. The solution specification is
+// validated before the enablement is started.
+//
+// The enablement only transfers ownership of the solution from EAM to LCCM
+// and sets the desired state in LCCM. The new desired state is not applied,
+// the solution system VMs are untouched.
+//
+// The following happens once the operation is started:
+//   - A removal of the corresponding agency in EAM is triggered.
+//
+// The following happens once the operation is completed:
+//   - The corresponding agency in EAM can no longer be controlled through the EAM API.
+//   - The management of the desired solution specification can be done only through vLCM. See Solutions
+//
+// Parameters:
```

Comment:
This is a JavaDoc style behavior and not used for Golang.

Reply:
ok